### PR TITLE
DM-42157: Add Python bindings for CoordinateSystem

### PR DESCRIPTION
**Pull request by taranu**

### CODE REVIEW

File: python/lib/include/pyimage.h [hunk 1]
```diff
@@ -120,16 +121,20 @@ void declare_image(py::module &m, std::string typestr) {
...
     .def("fill", &Class::fill)
     .def("get_value", &Class::get_value)
     .def("set_value", &Class::set_value)
     .def("get_value_unchecked", &Class::get_value_unchecked)
     .def("set_value_unchecked", &Class::set_value_unchecked)
     .def_property_readonly("size", &Class::size)
     .def_property_readonly("shape", &Class::shape)
+    .def(py::self == py::self)
+    .def(py::self != py::self)
+    .def(py::self += T())
```
Comment (fred3m) [inline python/lib/include/pyimage.h:137]: I've never seen any of these before. What do they do?

File: src/coordinatesystem.cc [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+            "dx1, dy2 = " + std::to_string(_dx1) + ", " + std::to_string(_dy2) + " !>0 or !finite"
+        );
+    }
+}
+
+CoordinateSystem::~CoordinateSystem() {}
+
+}  // namespace gauss2d
+
+#endif
```
Comment (fred3m) [inline src/coordinatesystem.cc:75]: Is this file really necessary? All of the methods are short, so I would just include them in the header, especially all of the one liners.

**Review by fred3m — APPROVED** [review]
LGTM, just a question and a comment.

File: python/lib/include/pyimage.h (hunk 1 above)
Comment (taranu) [reply python/lib/include/pyimage.h:137]: They're [various operators](https://pybind11.readthedocs.io/en/stable/advanced/classes.html#operator-overloading). The (in)equality ones are particularly strange to look at, but it is what it is.

File: src/coordinatesystem.cc (hunk 2 above)
Comment (taranu) [reply src/coordinatesystem.cc:75]: I think I've been fairly consistent about keeping implementations out of headers where possible (obviously not for template classes). It's more verbose but keeps the header neater.
